### PR TITLE
Issue #2281109: Porting 'Entity is of type (entity_is_of_type)' conditio...

### PR DESCRIPTION
**Pull request by ndewhurst**
...n to 8.x

See: d.o issue #2281109 - https://drupal.org/node/2281109
This creates an EntityIsOfType condition and an associated test that checks a simulated node's entity type.

### CODE REVIEW

File: src/Plugin/Condition/EntityIsOfType.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\TypedData\TypedDataManager $typed_data_manager
+   *   The typed data manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, TypedDataManager $typed_data_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $typed_data_manager);
```
Comment (fubhy) [inline src/Plugin/Condition/EntityIsOfType.php]: This constructor redundant.

File: src/Plugin/Condition/EntityIsOfType.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  public function summary() {
+    return $this->t('Entity is of type');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    // Load the entity to evaluate.
+    $providedEntity = $this->getContextValue('entity');
```
Comment (fubhy) [inline src/Plugin/Condition/EntityIsOfType.php]: We tend to not have camelCased variables. Our class fields are usually camelCased, variables in methods not though. :)

File: src/Tests/Condition/EntityIsOfTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Tests\Condition\EntityIsOfTypeTest.
+ */
+
+namespace Drupal\rules\Tests\Condition;
+
+use Drupal\Core\Path;
```
Comment (fubhy) [inline src/Tests/Condition/EntityIsOfTypeTest.php]: This does not seem to be in use.

File: src/Plugin/Condition/EntityIsOfType.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Condition\EntityIsOfType.
+ */
+
+namespace Drupal\rules\Plugin\Condition;
+
+use Drupal\Core\TypedData\TypedData;
```
Comment (fubhy) [inline src/Plugin/Condition/EntityIsOfType.php]: This does not seem to be in use.

File: src/Plugin/Condition/EntityIsOfType.php (hunk 1 above)
Comment (ndewhurst) [inline src/Plugin/Condition/EntityIsOfType.php]: Totally

File: src/Plugin/Condition/EntityIsOfType.php (hunk 2 above)
Comment (ndewhurst) [inline src/Plugin/Condition/EntityIsOfType.php]: Ah, good to know. I usually use camelCase to distinguish "objects" from other vars. I'll change this.

File: src/Plugin/Condition/EntityIsOfType.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    // Load the entity to evaluate.
+    $provided_entity = $this->getContextValue('entity');
+    // Retrieve the entity type specified as part of this condition.
+    $specified_type = $this->getContextValue('type');
+    // Retrieve the type of the evaluated entity.
+    $entity_type = $provided_entity->getEntityTypeId();
+    // Check to see whether the entity's type matches the specified value.
+    return $entity_type == $specified_type;
+  }
+}
```
Comment (fubhy) [inline src/Plugin/Condition/EntityIsOfType.php:77]: Missing carriage return at end of file. Sorry.. Missed that the first time.

File: src/Plugin/Condition/EntityIsOfType.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    // Load the entity to evaluate.
+    $provided_entity = $this->getContextValue('entity');
+    // Retrieve the entity type specified as part of this condition.
+    $specified_type = $this->getContextValue('type');
+    // Retrieve the type of the evaluated entity.
+    $entity_type = $provided_entity->getEntityTypeId();
+    // Check to see whether the entity's type matches the specified value.
```
Comment (fubhy) [inline src/Plugin/Condition/EntityIsOfType.php:74]: Some of these comments are kinda redundant and actually make it harder to read imho.

File: src/Tests/Condition/EntityIsOfTypeTest.php [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    $condition = $this->conditionManager->createInstance('rules_entity_is_of_type')
+      ->setContextValue('entity', $node)
+      ->setContextValue('type', 'node');
+    $this->assertTrue($condition->execute());
+
+    // Then test with values that should evaluate FALSE.
+    $condition->setContextValue('type', 'taxonomy_term');
+    $this->assertFalse($condition->execute());
+  }
+}
```
Comment (fubhy) [inline src/Tests/Condition/EntityIsOfTypeTest.php:74]: Also missing carriage return at the end of the file. You can configure your IDE to do that for you automatically.

File: src/Tests/Condition/EntityIsOfTypeTest.php [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+   * Tests evaluating the condition.
+   */
+  public function testConditionEvaluation() {
+    // Test with a node of type "page."
+    $node = $this->nodeStorage->create([
+      'type' => 'page',
+    ]);
+
+    // Add the test node to our context as the evaluated entity, along with an
+    //  explicit entity type string.
```
Comment (fubhy) [inline src/Tests/Condition/EntityIsOfTypeTest.php:63]: There is one whitespace too much before "explicit"

**Comment by fubhy** [comment]
Merged. I fixed those outstanding comments and converted the test to a unit test upon merge.

Thanks for the code! :)

**Pull request closed without merging**